apps/sms/views/shared/js/system_message_handler.js

```diff
@@ -329,6 +329,11 @@
       // Conversation view to the user, we remove that notification from the
       // tray that causes the second system message with "clicked" set to false.
       if (!notification.clicked || !notification.data) {
+        // Force close the app for notification removal case when app is hidden
+        // and navigation state is invalid(early reject while initializing).
+        if (document.hidden && !Navigation.isValidCurrentState()) {
```

Comment:
my problem is that isValidCurrentState will only be true at the first toPanel... this happens after a mozApps.getSelf(), wait that the document is visible (`onceDocumentIsVisible`) (so this should always be true if document.hidden is false), and that we retrieve the message information with `MessageManager.get(id)`.

Too many possible races...

IMO the only way if you don't want to use the task runner is to add a state like a boolean "this.handlingNotification". This would be good for 2.5 but I'd like that we convert using the task runner post-2.5.